src/main/kotlin/org/opensearch/replication/task/index/IndexReplicationTask.kt

```diff
@@ -518,13 +518,18 @@ open class IndexReplicationTask(id: Long, type: String, action: String, descript
                     for (alias in toAdd) {
                         log.info("Adding alias ${alias.alias} from $followerIndexName")
                         // Copying writeIndex from leader doesn't cause any issue as writes will be blocked anyways
-                        request.addAliasAction(AliasActions.add().index(followerIndexName)
-                                .alias(alias.alias)
-                                .indexRouting(alias.indexRouting)
-                                .searchRouting(alias.searchRouting)
-                                .writeIndex(alias.writeIndex())
-                                .isHidden(alias.isHidden)
-                        )
+                        var aliasAction = AliasActions.add().index(followerIndexName)
```

Comment:
we should consider having validation checks after overall metadata update, obtain diff and publish the index stats. 
- Can we employ similar mechanism as restore for updating overall index metadata?

Reply:
Yes, we can enhance the status api to reflect the same . 

>  Can we employ similar mechanism as restore for updating overall index metadata? 

the current metadata sync is a condensed version for same. What is main delta in the mechanism you are referring to ? can we track these as part of other issue and take that forward ?

Reply:
> we should consider having validation checks after overall metadata update, obtain diff and publish the index stats.

Are you referring to comparing with leader again? If so, it may be end up with false positives as updates to metadata/alias on leader could be moving target. If not, then agree - it can be a simpler validation by comparing "expected" update which we can capture while building request object.

Reply:
> Are you referring to comparing with leader again? If so, it may be end up with false positives as updates to metadata/alias on leader could be moving target

No, i was referring to the response already obtained as part of the this update from the leader. This will ensure that the metadata diff we obtained from the leader(minus replication specific settings etc) is on par with the current updated index state on the follower. 

> What is main delta in the mechanism you are referring to ? can we track these as part of other issue and take that forward ?

I was referring to the diff on overall IndexMetadata itself but in the current form, the response obtained from the leader (minus repl metadata for the index which we want to override) and the response after the update from the follower index. 


Ideally, these two should be equal and any deviation from this should be counted towards `replication_metadata_failure` as part of the index stats.

Reply:
Opened issue to track the overall changes - https://github.com/opensearch-project/cross-cluster-replication/issues/498